scipy/integrate/dop.py

```diff
@@ -14,5 +14,5 @@ def __dir__():
 
 def __getattr__(name):
     return _sub_module_deprecation(sub_package="integrate", module="dop",
-                                   private_module="_dop", all=__all__,
+                                   private_modules="_dop", all=__all__,
```

Comment:
Now that we're calling it `private_modules` (plural), shouldn't we change the API to take a list directly? It's internal anyway.
```suggestion
                                   private_modules=["_dop"], all=__all__,
```

Reply:
Since this is just an internal function, I preferred the laziness of being able to pass a string or a list so that all of the existing function calls didn't have to be changed. This is then handled by:
```
if not isinstance(private_modules, list):
        private_modules = [private_modules]
```
If you feel strongly about it I can change it but I would be keen to avoid the refactoring unless there is a strong reason/opinion

Reply:
I'm aware. I noted this because you're already touching all the callsites to change the keyword, so then adding some brackets does not make a big difference IMO.

Even though it's an internal function, I'd like to have it a clean API. To my mind there are two options:
1. keep the keyword the same (`private_module`, no plural), but "secretly" accept lists (and keep the listification of string arguments as an implementation detail in `_sub_module_deprecation`)
1. change the keyword to plural (`private_modules`), and change input to lists directly

Personally I think 2. is cleaner, but I'd be OK with either.